### PR TITLE
Multi-domain Species Selection and Electrocat RMG prep

### PR DESCRIPTION
**Pull request by mjohnson541**
This fixes a number of bugs in multidomain species selection and some bugs related to operations that happen inside RMG, but not in normal RMS operations.

### CODE REVIEW

File: src/Interface.jl [hunk 1]
```diff
@@ -198,16 +198,19 @@ function upgradekinetics(rxns,domain1,domain2)
...
             kin = stickingcoefficient2arrhenius(rxn.kinetics,surfdomain.phase.sitedensity,length(rxn.reactants)-1,spc[1].molecularweight)
-            rxns[i] = ElementaryReaction(index=rxn.index,reactants=rxn.reactants,reactantinds=rxn.reactantinds,products=rxn.products,
+            newrxns[i] = ElementaryReaction(index=rxn.index,reactants=rxn.reactants,reactantinds=rxn.reactantinds,products=rxn.products,
                 productinds=rxn.productinds,kinetics=kin,radicalchange=rxn.radicalchange,reversible=rxn.reversible,pairs=rxn.pairs)
+        else
+            newrxns[i] = rxn
         end
     end
-    return rxns
+    return [rxn for rxn in newrxns]
```
Comment (hwpang) [inline src/Interface.jl:213]: Why is this needed? To convert type?

File: src/EdgeAnalysis.jl [hunk 2]
```diff
@@ -268,46 +268,46 @@ function processfluxes(sim::SystemSimulation,
...
                     break
                 end
             end
         end
         index += size(d.rxnarray)[2]
     end
-    for d in inters
-        for i = index:index+size(d.rxnarray)[2]
+    for d in sim.interfaces
+        for i = 1:size(d.rxnarray)[2]
```
Comment (hwpang) [inline src/EdgeAnalysis.jl:295]: This might fail for intlet/outlet type interfaces where it doesn't have rxnarray attributes? Do we need a if statement before this to check?

**Review by hwpang — CHANGES_REQUESTED** [review]
Thanks for the PR! Mostly seems good, with a couple of comments/questions.

File: src/Interface.jl (hunk 1 above)
Comment (mjohnson541) [reply src/Interface.jl:213]: Yep, it causes it to retype the array.

File: src/EdgeAnalysis.jl (hunk 2 above)
Comment (mjohnson541) [reply src/EdgeAnalysis.jl:295]: I've added an if-statement to check if the rxnarray property is there. If not it skips it.

**Comment by mjohnson541** [comment]
Okay, I think I fixed everything and all tests pass.

**Review by hwpang — APPROVED** [review]
Awesome with the newest changes. this PR looks good to me. I will merge it soon!
